httpclient/httpclient.go

```diff
@@ -270,6 +286,24 @@ func (c *Client) retryRequest(ctx context.Context,
 	return backoff.Retry(attempt, backoff.WithContext(bo, ctx))
 }
 
+func (c *Client) shouldBackoff() error {
```

Comment:
why is `shouldBackoff` communicated as an `error` not `bool`?

Reply:
so we don't have to then also convert the bool to an error where it is used